trigger/context.go

```diff
@@ -2,26 +2,60 @@ package trigger
 
 import (
 	"context"
+	"github.com/project-flogo/core/support"
+	"time"
 )
 
-type key int
+var handlerKey = "_ctx_handler_info"
 
-var handlerKey key
+var idGenerator, _ = support.NewGenerator()
```

Comment:
need to make sure newgen hasn't failed...

Reply:
Good catch @mellistibco   Fixed it with just panic,  as it is very unlikely to happens

Reply:
@lixingwang @mellistibco Should that be a panic? Can we probably retry or use timestamp instead in such failure?

Reply:
init step should be fine.  If the error happens it gonna be a big issue.